### PR TITLE
Adding flag for ModelCheckpoint to save model as state_dict

### PR DESCRIPTION
**Pull request by akurniawan**
Trying to address https://github.com/pytorch/ignite/issues/160 by adding flag to store the model as state_dict and keeping the old method intact.
I have never used pytest before so I don't know whether creating the dummy model is the right way, please let me know if it's not following any standard :)

### CODE REVIEW

File: ignite/handlers/checkpoint.py [hunk 1]
```diff
@@ -87,53 +93,62 @@ def __init__(self, dirname, filename_prefix,
...
-
             else:
                 tmp.close()
                 os.rename(tmp.name, path)
 
+    def _internal_save(self, obj, path):
+        if not self._save_as_state_dict:
+            torch.save(obj, path)
+        else:
+            torch.save(obj.state_dict(), path)
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py]: Maybe, we need to check if `obj` has `state_dict` function ?

**Comment by vfdev-5** [comment]
@akurniawan thanks, looks good. Just wondering why there are many changes with line breakings only?

**Comment by akurniawan** [comment]
@vfdev-5 I forgot to add new line in test file every time I write new test case 😅, sorry for that

**Comment by vfdev-5** [comment]
@akurniawan okay, actually I meant the line breakings like that : 
```python
-            raise ValueError("Exactly one of `save_interval`, or `score_function` "
-                             "arguments must be provided.")
+            raise ValueError(
+                "Exactly one of `save_interval`, or `score_function` "
+                "arguments must be provided.")
```
There is a bunch of modifications like that... Maybe we can keep the previous line width as it has already passed lint checks. What do you think ?

**Comment by akurniawan** [comment]
@vfdev-5 ah I see, that’s the linter on my vscode trying to prettify the code. Will fix that

File: tests/ignite/handlers/test_checkpoint.py [hunk 2]
```diff
@@ -133,33 +161,122 @@ def test_best_k_with_suffix(dirname):
...
+        save_as_state_dict=True)
+
+    engine.add_event_handler(Events.EPOCH_COMPLETED, handler, {name: model})
+    engine.run([0], max_epochs=4)
+
+    saved_model = os.path.join(dirname, os.listdir(dirname)[0])
+    load_model = torch.load(saved_model)
+
+    assert not isinstance(load_model, DummyModel)
+    assert isinstance(load_model, dict)
```
Comment (alykhantejani) [inline tests/ignite/handlers/test_checkpoint.py]: can we also check the loaded state dict matches the model.state_dict()

**Comment by alykhantejani** [comment]
LGTM if we fix revert the line break changes. Also can you add an additional check in the test that the loaded state_dict is the same as the models, then its good to merge. 

Thanks for the PR!

**Comment by alykhantejani** [comment]
Thanks @akurniawan!

**Pull request merged**